include/npc/npc.h

```diff
@@ -2,15 +2,50 @@
 #define _NPC_H
 
 #include "game-state/game_state_common.h"
+#include "game-state/item.h"
+#include "game-state/player.h"
 #include "playerclass/class_structs.h"
 #include "playerclass/class.h"
 #include "npc/dialogue.h"
 #include "npc/npc_battle.h"
 #include "npc/npc_move.h"
 #include "cli/util.h"
+#include "quests/quests_hash.h"
+#include "quests/quests_state.h"
+
 
 // NPC STRUCTURE DEFINITION ---------------------------------------------------
 
+typedef struct quest_ctx quest_ctx_t; // a forward declaration
+typedef struct npc_quest npc_quest_t; // forward declaration
+
+/* 
+ * A singular quest node for npc_quest_list_t 
+ * (provided by the Quest team)
+*/ 
+typedef struct npc_quest {
+   char *id;
+   convo_t *dialogue;
+   npc_quest_t *next;
```

Comment:
you should be able to avoid using a forward declaration by doing this
```suggestion
   struct npc_quest *next;
```

Reply:
Thank you!

Reply:
Actually I tried this, and it causes compiler warnings (incompatible pointer type) in the npc_task/quest_list_add and npc_task/quest_list_free functions and other functions that involve traversing the list.

Reply:
ahhh, makes sense.